tests/expected/foreign-function/ffi_ptr.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Check Kani handling of C-FFI function pointers when C-FFI support is disabled.
+
+extern "C" {
+    fn foreign(i: u32) -> u32;
+}
+
+fn call_on(input: u32, func: unsafe extern "C" fn(u32) -> u32) -> u32 {
+    unsafe { func(input) }
+}
+
+fn may_not_call(call: bool, input: u32, func: unsafe extern "C" fn(u32) -> u32) -> Option<u32> {
+    call.then(|| unsafe { func(input) })
+}
+
+#[kani::proof]
+fn check_fn_ptr_called() {
+    let input: u32 = kani::any();
+    assert_eq!(call_on(input, foreign), input);
```

Comment:
how can we always guarantee that the return o `call_on` is equal to `input`?

Reply:
This dummy proof is incomplete since we don't have the body of `foreign`. This would only work if foreign is an identity function.

Reply:
btw, I fixed the test here, but what I think we should do later is turn this test into an `expected` test and make sure we are emitting the right error.